include/ppx/grfx/grfx_format.h

```diff
@@ -255,11 +267,56 @@ struct FormatDesc
     // In case of packed or compressed formats, this field is invalid
     // and the offsets will be set to -1.
     FormatComponentOffset componentOffset;
+
+    // In chroma-based formats, there can be subsampling of chroma color components
+    // of an image, to reduce image size.
+    FormatChromaSubsampling chromaSubsampling;
+
+    // If true, this is a planar format that does not store all image components
+    // in a single block. E.G. YCbCr formats, where Cb and Cr may be defined in
+    // a separate plane than Y values, and have a different resolution.
+    bool isPlanar;
+};
+
+enum FormatPlaneChromaType
+{
+    FORMAT_PLANE_CHROMA_TYPE_UNDEFINED,
+    FORMAT_PLANE_CHROMA_TYPE_LUMA,
+    FORMAT_PLANE_CHROMA_TYPE_CHROMA,
+};
+
+struct FormatPlaneDesc
+{
+    struct Member
+    {
+        // Note: it's expected that only one bit would be set here. That being
```

Comment:
maybe add an assert to make sure there is only 1 bit set when we use this var?

Reply:
Swapped it out for a new enum